StyleCop.Analyzers/StyleCop.Analyzers/Lightup/LightupHelpers.cs

```diff
@@ -65,15 +65,17 @@ internal static bool CanWrapObject(object obj, Type underlyingType)
                 return false;
             }
 
-            ConcurrentDictionary<Type, bool> wrappedObject = SupportedObjectWrappers.GetOrAdd(underlyingType, _ => new ConcurrentDictionary<Type, bool>());
-
             // Avoid creating the delegate if the value already exists
-            bool canCast;
-            if (!wrappedObject.TryGetValue(obj.GetType(), out canCast))
+            if (!SupportedObjectWrappers.TryGetValue(underlyingType, out var wrappedObject))
+            {
+                wrappedObject = SupportedObjectWrappers.GetOrAdd(underlyingType, static _ => new ConcurrentDictionary<Type, bool>());
+            }
+
+            // Avoid creating the delegate and capture class
+            if (!wrappedObject.TryGetValue(obj.GetType(), out var canCast))
             {
-                canCast = wrappedObject.GetOrAdd(
-                    obj.GetType(),
-                    kind => underlyingType.GetTypeInfo().IsAssignableFrom(obj.GetType().GetTypeInfo()));
```

Comment:
📝 This line is the source of the capturing allocation (specifically, capturing a method parameter in a lambda forces the the capture to be allocated on _all_ paths and not just the one that creates this delegate).